internal/jimm/cloudcredential_test.go

```diff
@@ -1740,3 +1740,110 @@ func (s testCloudCredentialAttributeStore) GetOAuthSecret(ctx context.Context) (
 func (s testCloudCredentialAttributeStore) PutOAuthSecret(ctx context.Context, raw []byte) error {
 	return errors.E(errors.CodeNotImplemented)
 }
+
+func TestCopyCredential(t *testing.T) {
+	c := qt.New(t)
+
+	ctx := context.Background()
+
+	api := &jimmtest.API{
+		CheckCredentialModels_: func(context.Context, jujuparams.TaggedCredential) ([]jujuparams.UpdateCredentialModelResult, error) {
+			return []jujuparams.UpdateCredentialModelResult{}, nil
+		},
+		UpdateCredential_: func(context.Context, jujuparams.TaggedCredential) ([]jujuparams.UpdateCredentialModelResult, error) {
+			return []jujuparams.UpdateCredentialModelResult{}, nil
+		},
+	}
+
+	j := jimmtest.NewJIMM(c, &jimm.Parameters{
+		Dialer: &jimmtest.Dialer{
+			API: api,
+		},
+	})
+
+	svcAccId, err := dbmodel.NewIdentity("39caae91-b914-41ae-83f8-c7b86ca5ad5a@serviceaccount")
+	c.Assert(err, qt.IsNil)
+	c.Assert(j.Database.DB.Create(&svcAccId).Error, qt.IsNil)
+	svcAcc := openfga.NewUser(svcAccId, j.OpenFGAClient)
```

Comment:
nowhere in this test do we grant alice@canonical.com any sort of rights on the svcAcc, which worries me a bit. i think we should only allow svc account admins to update cloud credentials

Reply:
See my comment above.